.github/workflows/build-and-publish-beta-chart.yml

```diff
@@ -0,0 +1,94 @@
+name: Manual Release Beta Chart
+
+# Manual trigger only
+on:
+  workflow_dispatch:
+    inputs:
+      version:
+        description: 'Version to use for the release'
+        required: false
+        default: ''
+      branch:
+        description: 'Branch to use for the release'
+        required: true
+
+jobs:
+  build-and-publish-chart:
+    permissions:
+      contents: 'write'
+      packages: 'write'
+      id-token: write
+      actions: 'read'
+    runs-on: ubuntu-latest
+    steps:
+
+      # Step 1: Checkout and Setup
+      - name: Checkout Code
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.inputs.branch }}
+          fetch-depth: 0
+
+      - name: Set up Git
+        run: |
+          git config --local user.email "ops@cloudzero.com"
+          git config --local user.name "Automated CZ Release"
+      
+      - name: Fetch All Branches
+        run: git fetch --all
+
+      # Step 2: Helm Setup
+      - name: Install Helm
+        uses: azure/setup-helm@v3
+
+      - name: Build Dependencies
+        run: helm dependency update charts/cloudzero-agent/
+
+      # Step 3: Determine Version
+      - name: Get Github Tag Version
+        id: version
+        uses: flatherskevin/semver-action@v1
+        with:
+          incrementLevel: patch
+          source: tags
+
+      - name: Determine Chart Version
+        run: |
+          NEW_VERSION=${{ github.event.inputs.version || steps.version.outputs.nextVersion }}
+          echo "NEW_VERSION=$NEW_VERSION-beta" >> $GITHUB_ENV
+
+      - name: Update Chart Version
+        run: |
+          VERSION_LINE=$(awk '/version:/ && !done {print NR; done=1}' charts/cloudzero-agent/Chart.yaml)
+          sed -i ''$VERSION_LINE's/.*/version: ${{ env.NEW_VERSION }}/' charts/cloudzero-agent/Chart.yaml
+
+      # Step 4: Package and Commit Chart
+      - name: Package Chart
+        run: helm package charts/cloudzero-agent/ --destination .deploy
+
+      - name: Commit updated Chart.yaml
+        run: |
+          git add .
+          git commit -m "Update Chart.yaml to version ${{ env.NEW_VERSION }}"
+          git push origin ${{ github.event.inputs.branch }}
+          COMMIT_HASH=$(git rev-parse HEAD)
+          echo "COMMIT_HASH=${COMMIT_HASH}" >> $GITHUB_ENV
+        continue-on-error: true
+
+      # Step 5: Create GitHub Release
```

Comment:
My understanding is this is how the tag is actually published as a release: https://github.com/Cloudzero/cloudzero-charts/releases. Let me know if this will cause any issues?

Reply:
Correct - that is my understanding also. There are two parts to it:
1. Update the index to include the new zip file
2. Update the new zip file

Both must be present to make helm work right.

Reply:
Thanks - added 👍